### PR TITLE
Replaced incorrect aget access FIX #3067

### LINKED ISSUE #3067: Replace incorrect aget access
### User Story

As a developer, I want to rely on good dev practises

### Description

*Type*: Feature

`aget` is only supposed to be used for `array` access and not `object fields` access. Replace all incorrect usages with proper access (to be defined).

### PR DESCRIPTION
**Pull request by ghost**
fixes #3067

### Summary: Replaced incorrect aget access

status: ready

### CODE REVIEW

File: src/status_im/data_store/realm/schemas/account/v6/contact.cljs [hunk 1]
```diff
@@ -37,7 +37,7 @@
   (let [new-contacts (.objects new-realm "contact")]
     (dotimes [i (.-length new-contacts)]
       (let [contact (aget new-contacts i)
-            id      (aget contact "whisper-identity")]
+            id      (.-whisper-identity contact)]
```
Comment (jeluard) [inline src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: I am skeptical here. Is it valid JavaScript?

Comment (yenda) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: why here in particular ?

Comment (jeluard) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: middlescore in JavaScript property names are an issue I believe

Comment (yenda) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: So it should also be a problem for all the other properties with middlescore ?

Comment (jeluard) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: Right

Comment (ghost) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: @jeluard @yenda Please check this Pastebin with my experiments on the question! I am very surprised how did it work before...
https://pastebin.com/JvhjJZNq
Unless our Figwheel REPL environment is different than the runtime...

Comment (jeluard) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: @poiga WDYT is the safe option?

Comment (jeluard) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: @pablodip I think you had some thoughts on that point?

Comment (pablodip) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: It seems that dashes [are not supported](https://stackoverflow.com/questions/5516106/are-dashes-allowed-in-javascript-property-names) in property names in Javascript. Not sure if ClojureScript converts property names to strings in those cases.

The safest and clearer option would probably be to access properties with dashes through `goog.object`.

Comment (poiga) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: I understand. but how did this `(aget contact "whisper-identity")` even work before?

Comment (jeluard) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: `aget` is supposed to work, it's using `object["field"]` JS semantic (not `object.field`)

Comment (dmitryn) [reply src/status_im/data_store/realm/schemas/account/v6/contact.cljs]: ClojureScript doesn't convert property names, it just silently returns nil.
```
(set! js/foo (js-obj "foo-bar" 1))
(aget js/foo "foo-bar") ;; => 1
(.-foo-bar js/foo) ;; => nil
```

`goog.object/get` should be used instead.

**Review by dmitryn — CHANGES_REQUESTED** [review]
`(.-key-type message)` is not a valid call, always returns nil. Please fix all calls where hyphened property names are used.

File: src/status_im/utils/notifications.cljs [hunk 2]
```diff
@@ -20,7 +20,7 @@
   (.requestPermissions (.-default rn/react-native-fcm)))
 
 (defn get-fcm-token []
-    (-> (.getFCMToken (aget rn/react-native-fcm "default"))
+    (-> (.getFCMToken (.-default rn/react-native-fcm))
```
Comment (poiga) [inline src/status_im/utils/notifications.cljs]: this one is very consistent with the one above.

**Comment by poiga** [comment]
@jeluard @yenda @dmitryn please check now!

File: src/status_im/utils/notifications.cljs (hunk 2 above)
Comment (dmitryn) [reply src/status_im/utils/notifications.cljs]: @poiga is there a reason you're mixing object/get and direct property access?  (even for non-hyphened properties). 95% of the time you're using object/get, but in some places like this one direct property call.

Comment (poiga) [reply src/status_im/utils/notifications.cljs]: @dmitryn can you see on the line 20 in the same file is the exact string - so I just wanted to keep it consistent  and readable.

Comment (dmitryn) [reply src/status_im/utils/notifications.cljs]: @poiga afaik, there are tradeoffs of using `(goog.object/get obj "prop")` or `(.-prop obj)`. Latter can be optimized by cljs compiler to reduce bundle size and obfuscate code in `:advanced` mode, but externing would be required, more info here https://github.com/gphilipp/react-native-with-clojurescript#building

We don't care much about obfuscating as we're open-source. Not sure about bundle size though, reducing its size might get us some benefits.

Overall, i think we should use dot notation everywhere except hyphened properties (where goog.object/get could be used).

**Pull request closed without merging**